osidb/tests/endpoints/test_endpoints.py

```diff
@@ -405,7 +405,7 @@ def test_flaw_components_create(self, auth_client, test_api_uri):
             "mitigation": "mitigation",
             "embargoed": False,
             "unembargo_dt": "2000-1-1T22:03:26.065Z",
-            "component": "curl",
+            "components": ["curl"],
```

Comment:
This updates a test checking the deprecated functionality, so the test should be removed :). And you can also remove the whole `TestDeprecationHandling` class.